[CDE-562][CLEANUP]

afrjorge opened the pull request.

afrjorge commented:
@pamval please review, should be merged along with webdetails/cde#434

pamval commented on cdf-core/cdf/js-lib/require-dashboard/dashboard.js:
```diff
@@ -0,0 +1,88 @@
+/*!
```
This doesn't seem documented according to jsdocs rules. Please review

afrjorge commented:
@pamval please review, added documentation for the dashboard loader plugin

afrjorge commented:
@pamval please review, renamed the "dashboard" loader plugin to "dash" to avoid confusion with the Dashboard class, and updated more documentation, should me merged along with webdetails/cde#435
